extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/utility.js

```diff
@@ -0,0 +1,143 @@
+angular.module('PaperUI.services').factory('util', function($filter, dateTime) {
+    return {
+        getItemStateText : function(item) {
+            if (item.state === 'NULL' || item.state === 'UNDEF') {
+                return '-';
+            }
+            if (item.stateDescription != null && item.stateDescription.options.length > 0) {
+                for (var i = 0; i < item.stateDescription.options.length; i++) {
+                    var option = item.stateDescription.options[i]
+                    if (option.value === item.state) {
+                        return option.label
+                    }
+                }
+            }
+            var state = item.type === 'Number' ? parseFloat(item.state) : item.state;
+
+            if (item.type === 'DateTime') {
+                var dateArr = item.state.split(/[^0-9]/);
+                var date;
+                if (dateArr.length > 5) {
+                    date = new Date(dateArr[0], dateArr[1] - 1, dateArr[2], dateArr[3], dateArr[4], dateArr[5]);
+                }
+                if (!date) {
+                    return '-';
+                }
+                if (item.stateDescription && item.stateDescription.pattern) {
+                    return timePrint(item.stateDescription.pattern, date, dateTime);
+                } else {
+                    return $filter('date')(date, "dd.MM.yyyy HH:mm:ss");
+                }
+            } else if (!item.stateDescription || !item.stateDescription.pattern) {
+                return state;
+            } else {
+                return sprintf(item.stateDescription.pattern, state);
+            }
+        }
+    }
+});
+function timePrint(pattern, date, dateTime) {
```

Comment:
Defining the function here will expose it to the global scope. Please move the function definition into the angular.module.factory() { <here> } block.